drivers/bluetooth/hci/userchan.c

```diff
@@ -93,6 +95,61 @@ static struct net_buf *get_rx(const uint8_t *buf)
 	return NULL;
 }
 
+/**
+ * @brief Decode the length of an HCI H4 packet
+ * @details Decodes packet length according to Bluetooth spec v5.2 Vol 4 Part E
+ * @param packet_type	Type of packet according to Bluetooth spec v5.2 Vol 4, Part A Table 2.1
+ * @param packet_buff	Pointer to packet buffer
+ */
+static uint16_t decode_hci_packet_length(uint8_t packet_type, const uint8_t *packet_buff)
+{
+	uint16_t payload_size = 0;
+	uint8_t header_size = 0;
+	const uint8_t packet_type_len = sizeof(packet_type);
+
+	switch (packet_type) {
+	case H4_CMD:
+		const struct bt_hci_cmd_hdr *hdr = &packet_buff[sizeof(packet_type)];
+
+		/* Parameter Total Length */
+		payload_size = hdr->param_len;
+		header_size = BT_HCI_CMD_HDR_SIZE;
+		break;
+	case H4_ACL:
+		const struct bt_hci_acl_hdr *hdr = &packet_buff[sizeof(packet_type)];
+
+		/* Data Total Length */
+		payload_size = sys_get_le16(&hdr->len);
```

Comment:
This generates unnecessarily complicated code. Just do `size = sys_le16_to_cpu(hdr->len);`. The structs are packed, so accessing the struct members generates automatically alignment-safe code (if needed by the CPU architecture).

Reply:
Does the packed attribute deal with endianness ? Thats the reason I thought parts like these ones are needed in the current bluetooths stack 

https://github.com/zephyrproject-rtos/zephyr/blob/2d65acca3a76b968657c2e796512558e93f2ed23/drivers/bluetooth/hci/h4.c#L127C3-L127C3

https://github.com/zephyrproject-rtos/zephyr/blob/2d65acca3a76b968657c2e796512558e93f2ed23/drivers/bluetooth/hci/h4.c#L140C2-L140C2

Reply:
I changed the other comment reviews accordingly but this one is still is not clear to me.

Could you be more specific on your suggestion? 

>  Just do `size = sys_le16_to_cpu(hdr->len)`

I understood either:

1. Should I change the name of the variable `payload_len` to `size` ?
2. Should I obtain the size of the payload via `size = sys_le16_to_cpu(hdr->len);` before the switch case? Its a trivial change and only the edge case for the struct `bt_hci_cmd_hdr` has to be taken into account (`param_len`).


Sorry for the misunderstanding on my part, looking forward for your comments.

Reply:
Ok, I think I overthought the comment. Changed to using `sys_le16_to_cpu` instead of `sys_get_le16` and the lenght member of the respective hci header struct instead of the address.